quantum/pointing_device/pointing_device_auto_mouse.c

```diff
@@ -0,0 +1,207 @@
+/* Copyright 2021 Christopher Courtney, aka Drashna Jael're  (@drashna) <drashna@live.com>
+ * Copyright 2022 Alabastard
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "pointing_device_auto_mouse.h"
+
+/* needed variables for tracking auto mouse */
+static auto_mouse_context_t auto_mouse = {.config.layer = (uint8_t)AUTO_MOUSE_DEFAULT_LAYER};
+
+/* Handle checking if a keycode is a mouse button */
+static bool is_mouse_record(uint16_t keycode, keyrecord_t* record) {
+    // allow for keyboard to hook in and override if need be
+    if (is_mouse_record_kb(keycode, record)) return true;
+    switch (keycode) {
+        case KC_MS_UP ... KC_MS_WH_RIGHT:
+            return true;
+    }
```

Comment:
For this, it might be simpler to do this: 
```suggestion
    if (is_mouse_record_kb(keycode, record) || IS_MOUSEKEY(keycode)) {
        return true;
    }
```

`IS_MOUSEKEY` is defined in /quantum/keycode.h, and checks for any of the mousekey keycodes.